Pengaturan Hari Merah

gundambison opened the pull request:
Modul/fitur baru yaitu pencatatan hari libur. terkait issue #4323
Dari update akan otomatis mengisi sabtu dan minggu ke table hari_merah untuk 2 tahun kedepan, menambahkan menu yang dapat di akses oleh admin

dapat memilih tanggal yang akan dirubah dan di ubah menjadi tanggal merah atau mengembalikan sebagai tanggal/hari kerja.

![dashboard 2](https://user-images.githubusercontent.com/1597200/132610652-b0013625-42a3-4cea-81c1-b26fc11ca748.jpg)
dashboard

![tambah 2](https://user-images.githubusercontent.com/1597200/132610693-886a817f-433a-40c4-966d-756ccec6918e.jpg)
penambahan data. Tanggal dapat di klik

![edit 2](https://user-images.githubusercontent.com/1597200/132610730-4959f3da-92b3-4786-bce5-ef4d0bbaa33c.jpg)
Pengeditan data. Tanggal tidak dapat dirubah

g-bison committed:
setting hari merah

g-bison committed:
modul tanggal merah

g-bison committed:
* penambahan menu baru
* list tangal dalam bentuk table
* ada filter pencarian
* update secara ajax

g-bison committed:
menyesuaikan masukan

andifahruddinakas commented:
Kenapa selalu buat PR baru setiap ada masukan/riview????

nurisakbar commented:
terima kasih mas @gundambison , saya sudah review dan secara konsep sudah ok.
cuman ada beberapa masukand dari saya 
- tambahkan fitur untuk hapus data disamping button edit
- fitur pencarian manual saya rasa tidak di perlukan disini, karena fitur pencarian datatables saya rasa sudah cukup untuk mewakili hal tersebut
- untuk layout bisa sedikit dirapikan biar lebih bagus

sebagai gambaran bisa dilihat pada tampilan ini :
![Screen Shot 2021-09-10 at 08 44 25](https://user-images.githubusercontent.com/5483330/132785466-0087b5b5-b2ef-47c2-b7aa-59c2843cfcd8.png)

terima kasih

g-bison committed:
memisahkan template dengan js. Update tampilan agar sesuai dengan menu lainnya

g-bison committed:
pembuatan login kehadiran
membuat numeric keyboard buat table/hp
test login masuk

g-bison committed:
perbaikan datatable hari merah

g-bison committed:
menambahkan route halaman kehadiran depan
menambah model kehadiran
update login form
memisahkan js dari view ke folder terpisah
membuat model logika kehadiran

g-bison committed:
login kehadiran
sesuai mockup
penambahan field pada pamong
tipe login masih 1 dan 3, 2 dan 4 menyusul

g-bison committed:
Merge pull request #1 from gundambison/kehadiran

Kehadiran

g-bison committed:
perbaikan tampilan login
memberikan kondisi apabila tanggal merah
tampilan login hanya terdapat 3
update tampilan form pamong
update model pamong

g-bison committed:
Merge pull request #2 from gundambison/kehadiran

perlu diperbaiki dari sisi UI dan UX saat login

gundambison commented:
Berikut update yang dilakukan:
- tampilan login masuk. ada info hari ini libur jika tanggal merah . **Karena ini di development, maka form login tetap ditampilkan**
![login](https://user-images.githubusercontent.com/1597200/133879538-325104b9-4c40-406c-8099-af55a0fc637f.jpg)
- tampilan login dengan scan (_perlu di test dan masukannya_)
![login tipe 2 (scan)](https://user-images.githubusercontent.com/1597200/133879561-71d0c126-3f2a-45fb-aee2-14fd7793468e.jpg)
- tampilan login dengan keyboard.. ini ada di login 1 dan 3..  **keyboard hanya numpad** .. hal ini sama seperti pin yang hanya numpad
![login - keyboard](https://user-images.githubusercontent.com/1597200/133879565-564f1f40-3917-4a80-b108-b258630de8e3.jpg)
- tampilan saat pamong baru melakukan login.. dia punya waktu **sekitar 20 detik** sebelum halaman menutup.
![masuk absen hadir](https://user-images.githubusercontent.com/1597200/133879602-684c8ddf-10ed-41ba-9420-987046117aa7.jpg) 
- tampilan saat sudah melakukan absen. 
![login-berhasil](https://user-images.githubusercontent.com/1597200/133879619-768de70d-9890-4add-ada1-f955c058b624.jpg)
- tampilan login untuk absen pulang 
![masuk absen pulang](https://user-images.githubusercontent.com/1597200/133879658-f4a81ce6-210f-4c0d-8043-5b6ae2e2e09a.jpg)
- tampilan berhasil absen keluar
![login-keluar](https://user-images.githubusercontent.com/1597200/133879669-0393bff3-29bd-49b5-acb4-6d7eabe6159e.jpg)
-tampilan tambahan pada pamong.. **hanya muncul pada pamong yang bukan dari penduduk** karena pin dan tag_id diambil dari table berbeda (penduduk_mandiri)
![pamong](https://user-images.githubusercontent.com/1597200/133879681-6d509ee3-fb61-478c-9cfc-7e5a2f316a68.jpg)

kendala: 

- alamat login untuk kehadiran ini diletakkan dimana? apa tidak perlu.. biar dari shortcut browser/gawai?
- wording yang digunakan mohon diperbaiki
- alur untuk pelaporan dari penduduk
- mockup untuk pelaporan.. apabila sulit, bisa menggunakan excel

g-bison committed:
routes

g-bison committed:
perbaikan routes

andifahruddinakas commented:
Pak @nurisakbar, jika PR ini sudah ok, bisa digabungkan ke https://github.com/OpenSID/OpenSID/tree/kehadiran-perangkat-desa
